ts/features/payments/checkout/store/reducers/index.ts

```diff
@@ -133,9 +134,18 @@ const reducer = (
         ...state,
         userWallets: pot.some(action.payload),
         selectedWallet: O.fromNullable(
-          action.payload?.wallets?.reduce((acc, curr) =>
-            acc.updateDate > curr.updateDate ? acc : curr
-          )
+          action.payload?.wallets
+            ?.filter(
+              wallet =>
+                wallet.clients.IO.status === WalletClientStatusEnum.ENABLED
+            )
+            .reduce((acc, curr) =>
+              acc.clients.IO?.lastUsage &&
+              curr.clients.IO?.lastUsage &&
+              acc.clients.IO.lastUsage > curr.clients.IO.lastUsage
+                ? acc
+                : curr
+            )
```

Comment:
We are using this filter + reduce function in different parts of the app, we should avoid code duplication by using a common function. Something like this:

```suggestion
        pipe(
          O.fromNullable(action.payload.wallets),
          O.chain(getLatestUsedWallet)
        )
```
Having this utility function:

```typescript
import * as RA from "fp-ts/lib/ReadonlyArray";
// ....

const getLatestUsedWallet = (
  wallets: ReadonlyArray<WalletInfo>
): O.Option<WalletInfo> =>
  pipe(
    wallets,
    RA.filter(
      wallet => wallet.clients.IO.status === WalletClientStatusEnum.ENABLED
    ),
    RA.reduce<WalletInfo, WalletInfo | undefined>(undefined, (acc, curr) =>
      acc?.clients.IO?.lastUsage &&
      curr.clients.IO?.lastUsage &&
      acc.clients.IO.lastUsage > curr.clients.IO.lastUsage
        ? acc
        : curr
    ),
    O.fromNullable
  );
  ``` 
  
  What do you think?

Reply:
Thanks for the useful hint! Now it is better!

Addressed into 7cde286605085d5983c76930abb4fbe3ca0fbd22